api/service.py

```
@@ -293,6 +293,87 @@ def get_current_username(credentials: HTTPBasicCredentials = Depends(security)):
+def process_telemetry_batch(telemetry_list: list) -> dict:
+    """Process a batch of telemetry data and return aggregated results."""
+    processed_count = 0
+    anomalies_detected = 0
+
+    for telemetry in telemetry_list:
+        try:
+            # Process individual telemetry (extracted from submit_telemetry logic)
+            processed_count += 1
+
+            # Check for anomalies
+            anomaly_score = anomaly_detector.detect_anomaly(telemetry)
```

Ruff (0.14.10)

Undefined name `anomaly_detector`

(F821)